lib/client/api/sharing.js

```diff
@@ -175,17 +189,140 @@ module.exports = function (jiffClient) {
+   *                          This must be one of 3 cases:
+   *                          1. null:                       open all shares to all parties.
+   *                          2. array of numbers:           open all shares to all the parties specified in the array.
+   *                          3. array of array of numbers:  open share with index i to the parties specified
+   *                                                         in the nested array at parties[i]. if parties[i] was null,
+   *                                                         then shares[i] will be opened to all parties.
+   * @param {string|number|object} [op_ids=auto_gen()] - an optional mapping that specifies the ID/Tag associated with each
+   *                                        open message sent. Since open_array involves sending many messages per party,
+   *                                        this parameter only specifies the BASE OPERATION ID. Each message sent will
+   *                                        have this base id attached to it concatenated to a counter.
+   *                                        If this is an object, then it should map an id of a receiving parties
+   *                                        to the base op_id that should be used to tag the messages sent to that party.
+   *                                        Parties left unmapped by this object will get an automatically generated id.
+   *                                        If this is a number/string, then it will be used as the base id tagging all messages
+   *                                        sent by this open to all parties.
+   *                                        You can safely ignore this unless you have multiple opens each containing other opens.
+   *                                        In that case, the order by which these opens are executed is not fully deterministic
+   *                                        and depends on the order of arriving messages. In this case, use this parameter
+   *                                        with every nested_open, to ensure ids are unique and define a total ordering on
+   *                                        the execution of the opens (check implementation of slt for an example).
+   * @returns {promise} a (JQuery) promise to ALL the open values of the secret, the promise will yield
+   *                    a 2D array of values, each corresponding to the given share in the shares parameter
+   *                    at the same index. In the case where different values are opened to different parties, the order
+   *                    of the values will be preserved, but not the indices, there will be no blanks in the resulting arrays,
+   *                    the first share that is opened to this party will appear at index [0], even if it was not initially
+   *                    at [0].
+   * @throws error if some shares does not belong to the passed jiff instance.
```

Comment:
yeah I think we should us this and share_array in the demos where applicable, and not use _ND_ version of it when we do not need to.